### PR TITLE
New cite-seq-count wrapper

### PR DESCRIPTION
**Pull request by lldelisle**
Hi,
This is a wrapper of cite-seq-count. This tool can be used in combination of STAR-solo to analyze CellPlex from Chromium (instead of Cell Ranger).

FOR CONTRIBUTOR:
* [X] - I have read the [CONTRIBUTING.md](https://github.com/galaxyproject/tools-iuc/blob/master/CONTRIBUTING.md) document and this tool is appropriate for the tools-iuc repo.
* [X] - License permits unrestricted use (educational + commercial)
* [X] - This PR adds a new tool or tool collection
* [ ] - This PR updates an existing tool or tool collection
* [ ] - This PR does something else (explain below)

### CODE REVIEW

File: tools/cite-seq-count/cite-seq-count.xml [hunk 1]
```diff
@@ -0,0 +1,526 @@
+<tool name="CITE-seq-Count" id="cite-seq-Count" version="@TOOL_VERSION@+galaxy@VERSION_SUFFIX@">
+    <description>This script counts matching antibody tags from paired fastq files.</description>
```
Comment (bgruening) [inline tools/cite-seq-count/cite-seq-count.xml:2]: can we shorten this?

File: tools/cite-seq-count/cite-seq-count.xml [hunk 2]
```diff
@@ -0,0 +1,526 @@
...
+            <output name="output_matrix_filtered">
+                <assert_contents>
+                    <has_line_matching expression="[1-2]\s+[1-4]\s+10"/>
+                    <has_n_lines n="9"/>
+                </assert_contents>
+            </output>
+        </test>
+    </tests>
+    <help><![CDATA[
+  CITE-seq-Count is a program that outputs umi and read counts from raq fastq CITE-seq or hashing data.
```
Comment (bgruening) [inline tools/cite-seq-count/cite-seq-count.xml:516]:
```suggestion
  CITE-seq-Count is a program that outputs UMI and read counts from raw FASTQ CITE-seq or hashing data.
```

File: tools/cite-seq-count/cite-seq-count.xml [hunk 3]
```diff
@@ -0,0 +1,526 @@
...
+    #set $r1 = $input_types.input_collection.forward
+    #set $r2 = $input_types.input_collection.reverse
+    #set $reads1 = $r1
+    #set $reads2 = $r2
+#end if
+CITE-seq-Count 
+--threads \${GALAXY_SLOTS:-4}
+--read1 $reads1
+--read2 $reads2
+--tags $tags
```
Comment (bgruening) [inline tools/cite-seq-count/cite-seq-count.xml:59]: single quotes

File: tools/cite-seq-count/cite-seq-count.xml [hunk 4]
```diff
@@ -0,0 +1,526 @@
...
+    #for $r1, $r2 in zip($input_types.input1, $input_types.input2):
+        #silent $reads1.append(str($r1))
+        #silent $reads2.append(str($r2))
+    #end for
+    #set $reads1 = ','.join($reads1)
+    #set $reads2 = ','.join($reads2)
+#elif str($input_types.use) == "list_paired":
+    #set $r1 = $input_types.input_collection.forward
+    #set $r2 = $input_types.input_collection.reverse
+    #set $reads1 = $r1
```
Comment (bgruening) [inline tools/cite-seq-count/cite-seq-count.xml:52]: is $r1 and $r2 needed here, you can remove the above two lines?

File: tools/cite-seq-count/cite-seq-count.xml [hunk 5]
```diff
@@ -0,0 +1,526 @@
...
+    #set $reads2 = $r2
+#end if
+CITE-seq-Count 
+--threads \${GALAXY_SLOTS:-4}
+--read1 $reads1
```
Comment (bgruening) [inline tools/cite-seq-count/cite-seq-count.xml:57]: I think those paths are not single-quoted - in both cases.

File: tools/cite-seq-count/cite-seq-count.xml [hunk 6]
```diff
@@ -0,0 +1,526 @@
...
+--cell_barcode_last_base $params.cell_barcode_last_base
+--umi_first_base $params.umi_first_base
+--umi_last_base $params.umi_last_base
+#end if
+--bc_collapsing_dist $bc_collapsing_dist
+--umi_collapsing_dist $umi_collapsing_dist
+$no_umi_correction
+--expected_cells $expected_cells
+#if $whitelist:
+--whitelist $whitelist
```
Comment (bgruening) [inline tools/cite-seq-count/cite-seq-count.xml:82]: no single quotes

**Comment by lldelisle** [comment]
Thank you @bgruening for the review. I think I resolved all.

**Review by bgruening — APPROVED** [review]
Awesome!

**Comment by lldelisle** [comment]
Thank you :heart_eyes:

**Pull request merged**